src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -224,7 +236,16 @@ private static bool IsDeclarationStartContext(List<SyntaxBase> matchingNodes, in
             return false;
         }
 
-        private static bool IsPropertyNameContext(List<SyntaxBase> matchingNodes, (ObjectSyntax? node, int index) objectInfo)
+        private static bool IsPropertyAccessContext(List<SyntaxBase> matchingNodes, (PropertyAccessSyntax? node, int index) propertyAccessInfo)
+        {
+            // if we're accessing properties, we have the following matching node types at the end of the list:
+            // PropertyAccessSyntax, IdentifierSyntax, Token(TokenType.Identifier)
+            return propertyAccessInfo.node != null &&
+                   SyntaxMatcher.IsTailMatch<PropertyAccessSyntax, IdentifierSyntax, Token>(matchingNodes, (propertyAccess, identifier, token) =>
```

Comment:
Testing out a new approach to writing these rules. If it works out, I will replace the other rules with this approach over time.